Resources/config/serializer.xml

```diff
@@ -17,6 +17,7 @@
         <!-- Normalizes FormInterface when using the symfony serializer -->
         <service id="fos_rest.serializer.form_error_normalizer" class="FOS\RestBundle\Serializer\Normalizer\FormErrorNormalizer" public="false">
             <argument type="service" id="translator" />
+            <argument>%validator.translation_domain%</argument>
```

Comment:
using DI paramters as extension point is not a good solution. we do not need a paramter for this. having a constructor argument on the FormErrorHandler is enough as you can then simply decorate the service and change the translation domain.